gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/EntityInteractionsFetcher.java

```diff
@@ -164,16 +164,18 @@ public Set<EntityKey> fetchInteractionsIdsIfNecessary(
                         () -> executeQueryRequest(context, e), this.queryExecutor))
             .collect(Collectors.toUnmodifiableList());
 
-    return queryRequestCompletableFutures.stream()
-        .map(CompletableFuture::join)
-        .map(response -> parseInteractionIdsResponse(entitiesRequest, response))
-        .flatMap(Collection::stream)
-        .collect(Collectors.toUnmodifiableSet());
+    LinkedHashSet<EntityKey> entityKeys =
+        queryRequestCompletableFutures.stream()
+            .map(CompletableFuture::join)
+            .map(response -> parseInteractionIdsResponse(entitiesRequest, response))
+            .flatMap(Collection::stream)
+            .collect(Collectors.toCollection(LinkedHashSet::new));
```

Comment:
nit - are you trying to use the set just to make sure there aren't dupes? If your goal is for it to be both ordered and no dupes, it would be clearer to just do:
```
.distinct()
.collect(Collectors.toUnmodifiableList());
```